parsing/dialects/intermediate/dialects/dialect1/parserImpls.ftl

```diff
@@ -5741,3 +5740,124 @@ SqlSelectInto SqlSelectInto() :
             fromClause, whereClause);
     }
 }
+
+SqlCall FirstLastValue() :
+{
+    final SqlNode value;
+    final SqlNode over;
+    final List<SqlNode> args = new ArrayList<SqlNode>();
+    SqlNullTreatmentModifier nullTreatmentModifier = null;
+    final SqlAggFunction function;
+    final SqlCall firstLastCall;
+}
+{
+    (
+        <FIRST_VALUE> {
+            function = SqlStdOperatorTable.FIRST_VALUE;
+        }
+    |
+        <LAST_VALUE> {
+            function = SqlStdOperatorTable.LAST_VALUE;
+        }
+    )
+    <LPAREN>
+    value = CompoundIdentifier() { args.add(value); }
+    [
+        (
+            <IGNORE> {
+                nullTreatmentModifier = new SqlNullTreatmentModifier(getPos(),
+                    SqlKind.IGNORE_NULLS);
+            }
+        |
+            <RESPECT> {
+                nullTreatmentModifier = new SqlNullTreatmentModifier(getPos(),
+                    SqlKind.RESPECT_NULLS);
+            }
+        )
+        <NULLS> { args.add(nullTreatmentModifier); }
+    ]
+    {
+            firstLastCall = function.createCall(getPos(), args);
+    }
+    <RPAREN>
+    <OVER>
+    over = WindowSpecification()
+    {
+        return SqlStdOperatorTable.OVER.createCall(getPos(),
+            firstLastCall, over);
+    }
+}
+
+/**
+ * Parses a reserved word which is used as the name of a function.
+ */
+SqlIdentifier ReservedFunctionName() :
```

Comment:
Is it safe to assume this new function is overriding a function in parsing/parserImpls.ftl? Maybe a piece of comment would help clarify.

Reply:
You are correct, it is overriding. I'll add a comment.

Reply:
A comment can't hurt. But I'm also not sure it's entirely necessary. We're already overloading lots of functions in Dialect1 that don't have comments (e.g. SqlSelect).

No strong preference from me on this.

Reply:
I'll add a comment specifying that we removed FIRST_VALUE and LAST_VALUE, otherwise it may be difficult to tell what the difference is.

Reply:
The issue with a comment like that is that it will immediately grow out of date (or be onerous to keep updated) as soon as we start adding/removing other keywords from here.

Reply:
I think the comments/annotations added in https://github.com/googleinterns/calcite/pull/215 should be sufficient to help a developer understand what's being overloaded and what's not.

Reply:
Ok, SGTM.